app/features/inspector/inspector.tsx

```diff
@@ -0,0 +1,102 @@
+import nextPageViewerSearch from "app/search/flows/next-page-viewer-search"
+import {isEqual} from "lodash"
+// @ts-strict
+import React, {ComponentType, ReactNode} from "react"
+import {useDispatch, useSelector} from "react-redux"
+import {ListChildComponentProps} from "react-window"
+import Viewer from "src/js/state/Viewer"
+import {zed} from "@brimdata/zealot"
+import {inspect} from "./inspect"
+import {List} from "./list.styled"
+import {Context, RowData} from "./types"
+
+type Props = {
+  height: number
+  width: number
+  values: zed.Value[]
+  defaultExpanded: boolean
+  expanded: Map<zed.Value, boolean>
+  setExpanded: (m: Map<zed.Value, boolean>) => void
+  onContextMenu: Context["onContextMenu"]
+}
+
+export function Inspector(props: Props) {
+  const dispatch = useDispatch()
+  const isFetching = useSelector(Viewer.getStatus) === "FETCHING"
+  const isIncomplete = useSelector(Viewer.getEndStatus) === "INCOMPLETE"
+  const {defaultExpanded, expanded, setExpanded} = props
+  const context: Context = {
+    indent: 0,
+    rows: [],
+    onContextMenu: props.onContextMenu,
+    nest() {
+      this.indent += 1
+    },
+    unnest() {
+      this.indent -= 1
+    },
+    push(render: ReactNode) {
+      this.rows.push({render, indent: this.indent})
+    },
+    isExpanded: (value: zed.Value) => {
+      if (expanded.has(value)) return expanded.get(value)
+      else return defaultExpanded
+    },
+    setExpanded: (value: zed.Value, bool: boolean) => {
+      const newMap = new Map(expanded.entries())
+      newMap.set(value, bool)
+      setExpanded(newMap)
+    }
+  }
+
+  for (const v of props.values) {
+    inspect({
+      ctx: context,
+      value: v,
+      field: null,
+      key: null,
+      last: true,
+      // @ts-ignore
+      type: v.type
+    })
+  }
+
+  return (
+    <List
```

Comment:
Would be nice to add an empty state here if there are no rows to render

Reply:
Good point.